webapp/src/js/actions/index.js

```diff
@@ -14,13 +13,52 @@ angular.module('inboxServices').factory('Actions',
         };
       }
 
+      function createSetEnketoErrorAction(error) {
+        return {
+          type: 'SET_ENKETO_ERROR',
+          payload: {
+            error: error
+          }
+        };
+      }
+
+      function createSetEnketoEditedStatusAction(edited) {
+        return {
+          type: 'SET_ENKETO_EDITED_STATUS',
+          payload: {
+            edited: edited
+          }
+        };
+      }
+
+      function createSetEnketoSavingStatusAction(saving) {
+        return {
+          type: 'SET_ENKETO_SAVING_STATUS',
+          payload: {
+            saving: saving
+          }
+        };
+      }
```

Comment:
This feels repetitive...

Are these always 1-to-1? If so shouldn't they just be inlined so you can easily see what you're returning?

Reply:
I'd love an ENUM right about now...

Reply:
This is indeed repetitive! Boilerplate is a lot of people's main gripe with redux. Some of this could be trimmed down though actually I'll do that now.

Do you mean 1-to-1 for example this function
```
function createSetEnketoSavingStatusAction(saving) {
        return {
          type: 'SET_ENKETO_SAVING_STATUS',
          payload: {
            saving: saving
          }
        };
      }
```
paired with this one?
```
setEnketoSavingStatus: function(saving) {
          dispatch(createSetEnketoSavingStatusAction(saving));
        },
```

If so then no they won't always be 1-to-1, for now you could definitely inline them but soon there'll be more complex functions dispatching several actions and the same action will be dispatched from several different functions.

Reply:
I guess we'll see how it plays out - we can inline it later if it makes sense to do so.